swap/htlc.go

```diff
@@ -81,13 +84,26 @@ func NewHtlc(version ScriptVersion, cltvExpiry int32,
 	hash lntypes.Hash, outputType HtlcOutputType,
 	chainParams *chaincfg.Params) (*Htlc, error) {
 
-	if version != HtlcV1 {
```

Comment:
This could have been made a switch already, to reduce total pr diff

Reply:
done